### PR TITLE
Switch ContentType to ES6 class structure and added Taxonomy, User an…

### PR DESCRIPTION
**Pull request by eroepken**
…d Base classes. (Base is abstract and cannot be instantiated.) Added JSDoc package.

### CODE REVIEW

**Comment by eroepken** [comment]
This is failing tests. Fixing now.

File: src/models/Taxonomy.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+class Taxonomy extends Base {
+
+	constructor() {
```
Comment (eroepken) [inline src/models/Taxonomy.js]: This is missing a call to the super class (Base).

File: src/models/User.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+class User extends Base {
+
+	constructor() {
```
Comment (eroepken) [inline src/models/User.js]: This is missing a call to the super class (Base).

File: src/models/ContentType.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+class ContentType extends Base {
+
+	constructor() {
```
Comment (eroepken) [inline src/models/ContentType.js]: This is missing a call to the super class (Base).

**Comment by eroepken** [comment]
I left comments on the places where the tests were failing so other people can learn as well what they can do to prevent test failures on syntax errors.

**Comment by eroepken** [comment]
If this is okay with you guys, I will merge it.

**Comment by Atomox** [comment]
Looks good to me.

**Pull request merged**